### PR TITLE
CRUD Airline

### PR DESCRIPTION
**Pull request by adibnajwan**

### CODE REVIEW

**Comment by adibnajwan** [comment]
semangat

File: src/routes/airline.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+export default (app) => {
+  const router = Router();
+  app.use('/airline', router);
+
+  router.post(
+    '/',
+    authMiddleware.isAuthorized,
+    authMiddleware.isAdmin,
+    airlineValidation.createAirlineValidation,
+    airlineMiddleware.checkAirlineNameExist,
```
Comment (norplys) [inline src/routes/airline.js]: double middleware

File: src/routes/airline.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    airlineMiddleware.checkAirlineNameExist,
+    airlineController.createAirline
+  );
+
+  router.get(
+    '/:id',
+    airlineMiddleware.checkAirlineById,
```
Comment (norplys) [inline src/routes/airline.js]: validate id params, take it from common middlewares

File: src/routes/airline.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    airlineController.getAirlineById
+  );
+
+  router.put(
+    '/:id',
+    authMiddleware.isAuthorized,
+    authMiddleware.isAdmin,
+    commonValidationMiddleware.validateIdParams,
+    airlineValidation.updateAirlineValidation,
+    airlineMiddleware.checkAirlineById,
```
Comment (norplys) [inline src/routes/airline.js]: Add check by code and name

File: src/controllers/airline.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+export async function deleteAirline(req, res) {
+  const { id } = req.params;
+
+  const airlineData = await airlineServices.deleteAirlineById(id);
+
+  res.status(200).json({
+    message: 'Airline deleted successfully',
+    data: airlineData
```
Comment (norplys) [inline src/controllers/airline.js]: remove data

**Review by norplys — CHANGES_REQUESTED** [review]

File: src/routes/airlines.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    authMiddleware.isAuthorized,
+    authMiddleware.isAdmin,
+    airlineValidation.createAirlineValidation,
+    airlineMiddleware.checkAirlineCodeOrNameExist,
+    airlineController.createAirline
+  );
+
+  router.get(
+    '/:id',
+    airlineMiddleware.checkAirlineById,
```
Comment (norplys) [inline src/routes/airlines.js:26]: validate id paramsnya

File: src/middlewares/airline.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import * as airlineServices from '../services/airline.js';
+import { HttpError } from '../utils/error.js';
+
+export async function checkAirlineCodeOrNameExist(req, res, next) {
+  const { name, code } = req.body;
+  const currentAirline = res.locals.airport;
```
Comment (norplys) [inline src/middlewares/airline.js]: its airline not airport

File: src/services/airline.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import * as airlineRepository from '../repositories/airline.js';
+
+export async function createAirline(payload) {
```
Comment (norplys) [inline src/services/airline.js:3]: await nya masukkin ke variabel data trus return datanya ( all of it )

**Review by norplys — CHANGES_REQUESTED** [review]

File: src/services/airline.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  return await airlineRepository.getAirlineByNameOrCode(name, code);
+}
+
+export async function updateAirlineById(id, payload) {
+  return await airlineRepository.updateAirline(id, payload);
+}
+
+export async function deleteAirlineById(id) {
+  return await airlineRepository.deleteAirline(id);
+}
```
Comment (norplys) [inline src/services/airline.js]: buat ke return data semua

File: src/repositories/airline.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+export function updateAirline(id, payload) {
+  return prisma.airline.update({
+    where: {
+      id
+    },
+    data: {
+      ...payload,
+      updatedAt: new Date()
```
Comment (norplys) [inline src/repositories/airline.js]: updateAt jan di update manual

File: src/routes/index.js [hunk 10]
```diff
@@ -17,4 +18,5 @@ export default (app) => {
   docs(router);
   aeroplanes(router);
   airports(router);
+  airline(router);
```
Comment (norplys) [inline src/routes/index.js]: s wkwkw

**Review by norplys — APPROVED** [review]
LGTM !

**Pull request merged**